editor/import/resource_importer_scene.cpp

```diff
@@ -2369,7 +2369,7 @@ Node *ResourceImporterScene::pre_import(const String &p_source_file, const HashM
 	ERR_FAIL_COND_V(!importer.is_valid(), nullptr);
 
 	Error err = OK;
-	HashMap<StringName, Variant> options_dupe = p_options;
+	const HashMap<StringName, Variant> &options_dupe = p_options;
```

Comment:
I'm unsure about this one, it does say `dupe`, so it is intended to be copied, but it is not modified down the line

For this and a number of other cases I'd instead suggest replacing references *to* it with `p_options`, or leave as is, not making a reference here, as either they should be passed directly, or copied, this case creates a single-use reference to a reference with no purpose

Reply:
Especially true for `Ref<T> foo = p_foo`, they should just be replaced as they are just strange code originally

Same would go for the COW types, like `String` and `Vector`

For `HashMap` that would require more checking as it is *not* a COW or refcounted type